### PR TITLE
Update DG (JY)

### LINKED ISSUE #503: Update DG - Implementation (JY)

### PR DESCRIPTION
**Pull request by jianyangg**
Updated implementation for register account in terms of:
* Sequence diagram
* Activity diagram
* Their descriptive text

Close #503

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -355,41 +344,46 @@ The format for the `register` command can be found [here](UserGuide.md#register)
...
-   and `Forget Password Answer` in the `register` command.
-2. If any of the fields is not provided, an error message with the correct command usage will be shown.
-3. If invalid command parameters are provided, an error message with the correct parameter format will be shown.
-4. If the `Password` and `Confirm Password` fields do not match, an error message will be shown.
-5. If the user is currently logged in, an error message will be shown.
-6. The `User` is then cross-referenced with the stored user in `Model` to check if there's already a user stored. If
+1. The user specifies the `USERNAME`{.swift}, `PASSWORD`{.swift}, `CONFIRM_PASSWORD`{.swift}, `SECRET_QUESTION`{.swift}
+   and `ANSWER`{.swift} in the `register`{.swift} command.
+1. If any of the fields is not provided, an error message with the correct command format will be shown.
+1. If command parameters provided are not within the acceptable value range, an error message with the correct command format will be shown.
```
Comment (B-enguin) [inline docs/DeveloperGuide.md:396]: Would value range sound abit wrong, since its not really a range?

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -355,41 +344,46 @@ The format for the `register` command can be found [here](UserGuide.md#register)
...
+1. Logic Manager calls the `AddressBookParser#parseCommand` with the full user input.
+1. The `AddressBookParser` parses the command word, creating an instance of `UserRegisterCommandParser` to parse the
    rest of the command.
-4. If all fields are present, it checks if password and confirm password match.
-5. If password and confirm password match, it creates an instance of `UserRegisterCommand`.
-6. `Logic Manager` executes `UserRegisterCommand` by calling `UserRegisterCommand#execute()`.
-7. `UserRegisterCommand` checks if a user is already registered by calling `Model#getStoredUser`.
-8. If no user is registered, `UserRegisterCommand` calls `Model#registerUser` to store the user. Login status is set to
+1. If not all fields are present, a `ParseException` is thrown saying incorrect command format.
+1. If all fields are present, it checks if `PASSWORD`{.swift} and `CONFIRM_PASSWORD`{.swift} match.
```
Comment (B-enguin) [inline docs/DeveloperGuide.md:418]: Would a check for password format be missing here?

**Review by B-enguin — COMMENTED** [review]
Some comments

**Review by Gabriel4357 — APPROVED** [review]
LGTM

**Pull request merged**